src/improvements/tasks/StateDiffDecoder.sol

```diff
@@ -0,0 +1,536 @@
+                uint256 offset = layout[i]._offset;
+                string memory oldValue;
+                string memory newValue;
+                if (kind.eq("bool")) {
+                    oldValue = toBool(_oldValue, offset);
+                    newValue = toBool(_newValue, offset);
+                } else if (kind.eq("address")) {
+                    oldValue = toAddress(_oldValue, offset);
+                    newValue = toAddress(_newValue, offset);
+                } else if (kind.contains("uint")) {
+                    oldValue = toUint(_oldValue, offset);
+                    newValue = toUint(_newValue, offset);
+                }
+
+                string memory label = layout[i]._label;
+                return DecodedSlot({kind: kind, oldValue: oldValue, newValue: newValue, summary: label, detail: ""});
+            }
+        }
+    }
+
+    /// @notice Given the path to a JSON file and a target address, returns the first chain ID and
+    /// contract name where the value equals the target.
+    function findContractByAddress(string memory filePath, address target)
+        internal
+        view
+        returns (uint256 l2ChainId_, string memory contractName_)
+    {
+        // Read the entire JSON file.
+        string memory jsonData = vm.readFile(filePath);
+
+        // Get all the top-level keys, which are the chain IDs
+        string[] memory chainIds = vm.parseJsonKeys(jsonData, "$");
+        for (uint256 i = 0; i < chainIds.length; i++) {
+            string memory chainId = chainIds[i];
+
+            // Get all the keys of the nested object under currentTopKey.
+            string memory key = string.concat("$.", chainId);
+            string[] memory contractNames = vm.parseJsonKeys(jsonData, key);
+            for (uint256 j = 0; j < contractNames.length; j++) {
+                string memory contractName = contractNames[j];
+
+                // Build the JSON path: e.g. ".10.Guardian"
+                string memory path = string.concat(".", chainId, ".", contractName);
+                address foundAddress = vm.parseJsonAddress(jsonData, path);
+
+                if (foundAddress == target) {
+                    // If the contract name ends with "Proxy", strip it.
+                    if (contractName.endsWith("Proxy")) {
+                        contractName = contractName.slice(0, bytes(contractName).length - 5);
+                    }
+                    // If the contract name is "OptimismPortal", change it to "OptimismPortal2".
+                    if (contractName.eq("OptimismPortal")) {
+                        contractName = "OptimismPortal2";
+                    }
+                    // We make a call to see if the contract is a Safe.
+                    if (isGnosisSafe(target)) {
+                        contractName = string.concat(contractName, " (GnosisSafe)");
+                    }
+
+                    return (vm.parseUint(chainId), contractName);
+                }
+            }
```

Comment:
@ElliotFriedman @blmalone @prat-gpt when we implement this decoded logging into the new version, we can replace this block with a lookup in the address registry using the `getAddressInfo` method that @prat-gpt added